src/UnitTests/ValueTransformers.cs

```diff
@@ -0,0 +1,247 @@
+﻿using Shouldly;
+using Xunit;
+
+namespace AutoMapper.UnitTests
+{
+    namespace ValueTransformers
+    {
+        public class BasicTransforming : AutoMapperSpecBase
+        {
+            public class Source
+            {
+                public string Value { get; set; }
+            }
+
+            public class Dest
+            {
+                public string Value { get; set; }
+            }
+
+            protected override MapperConfiguration Configuration { get; } = new MapperConfiguration(cfg =>
+            {
+                cfg.CreateMap<Source, Dest>();
+                cfg.ApplyTransform<string>(dest => dest + " is straight up dope");
+            });
+
+            [Fact]
+            public void Should_transform_value()
+            {
+                var source = new Source
+                {
+                    Value = "Jimmy"
+                };
+                var dest = Mapper.Map<Source, Dest>(source);
+
+                dest.Value.ShouldBe("Jimmy is straight up dope");
+            }
+        }
+
+        public class StackingTransformers : AutoMapperSpecBase
+        {
+            public class Source
+            {
+                public string Value { get; set; }
+            }
+
+            public class Dest
+            {
+                public string Value { get; set; }
+            }
+
+            protected override MapperConfiguration Configuration { get; } = new MapperConfiguration(cfg =>
+            {
+                cfg.CreateMap<Source, Dest>();
+                cfg.ApplyTransform<string>(dest => dest + " is straight up dope");
+                cfg.ApplyTransform<string>(dest => dest + "! No joke!");
```

Comment:
@lbargaoanu I gotcha, add the extension method to the list of value transformers rather than the root. I don't want to create a custom collection type.

@asherber I think we could add conditions to transforms that would allow for turning things off (or don't apply things globally).